confidence.py

```diff
@@ -147,8 +147,13 @@ def __init__(self, *sources, separator='.'):
         :param separator: the character(s) to use as the separator between keys
         """
         self._separator = separator
+        self._missing = missing
         self._root = self
 
+        if isinstance(self._missing, str):
+            self._missing = {'safe': NotConfigured,
+                             'raise': _NoDefault}[missing]
```

Comment:
Allowing non-strings as 'default' values is sneaky. The 'magic' values should maybe be an enum, with documentation stating that those values are sentinels and any other values will be the default for missing keys.

Reply:
Documentation pending, enum added.